controllers/deactivation/mapper.go

```diff
@@ -0,0 +1,53 @@
+package deactivation
+
+import (
+	"context"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type UserSignupToMasterUserRecordMapper struct {
+	client client.Client
+}
+
+var _ handler.Mapper = UserSignupToMasterUserRecordMapper{}
+var mapperLog = ctrl.Log.WithName("UserSignupToMasterUserRecordMapper")
+
+func (b UserSignupToMasterUserRecordMapper) Map(obj handler.MapObject) []reconcile.Request {
+	if userSignup, ok := obj.Object.(*toolchainv1alpha1.UserSignup); ok {
+		// look-up any associated MasterUserRecord using the UserSignup's Status.CompliantUsername value
+		mur := &toolchainv1alpha1.MasterUserRecord{}
+
+		if err := b.client.Get(context.TODO(), client.ObjectKey{
+			Namespace: userSignup.Namespace,
+			Name:      userSignup.Status.CompliantUsername,
+		}, mur); err != nil {
+			mapperLog.Error(err, "Could not find MasterUserRecord resource with name", "name",
+				userSignup.Status.CompliantUsername)
+			return nil
+		}
```

Comment:
May I know why do you get the MUR? As I see it - the mapper should only deal with mapping, which means that it should only return the `userSignup.Status.CompliantUsername`. If the MUR doesn't exist, then it's not the responsibility of the mapper, but of the controller.

Reply:
In addition to that, the fact that the MUR doesn't exist is not an error - it's a pretty valid use-case, so we should not log an error here.
If you want to filter out those UserSignups whose MUR doesn't exist, then you should use predicated for that, not a mapper.

Reply:
Thanks for the explanation, I've removed the MUR lookup.